prac_07/programming_language.py

```diff
@@ -0,0 +1,21 @@
+class ProgrammingLanguage:
+    """CP1403 programming language class"""
+
+    def __init__(self, name: str = "", typing: str = "", reflection: bool = False, year: int = "0000"):
+        """initialise a language type
+
+        Reflection: boolean, False = 'No', True = 'Yes'
+        """
+        self.name = name
+        self.typing = typing
+        self.reflection = reflection
+        self.year = year
+
+    def __is_dynamic__(self):
+        """returns True if programming language is dynamic (as opposed to static)"""
+        return "{}".format(True if self.typing == "Dynamic" else False)
```

Comment:
Doesn't this return a str? It should return a boolean. Drop the string formatting.

Reply:
changed to:
        return self.typing == "Dynamic"